staging/src/k8s.io/cloud-provider/plugins.go

```diff
@@ -116,7 +128,7 @@ func InitCloudProvider(name string, configFilePath string) (Interface, error) {
 		return nil, nil
 	}
 
-	if IsExternal(name) {
+	if IsExternal(name, false) {
```

Comment:
you should pass the correct value here.  Since you can return an error, once you update `IsExternal` to return an error, this call because natural.  I only see 8 or so callers to InitCloudProvider.

Reply:
@deads2k I don't think InitCloudProvider could be changed so simply. This method is in API used in every external CCM binary outside of k/k, and it makes no sense to pass a kubernetes only feature gate configuration to this method and forcing everyone to specifically `enableCloudProviders` after a rebase on k/k 1.22. And we can't use `utilfeature.DefaultFeatureGate.Enabled(features.DisableCloudProviders)` in this package too because it introduces dependency on core k8s and additionally will not pass the CI.

Reply:
If IsExternal requires the information, then I think it makes sense to pass the information.  You can change its name if you like, but as you have the code written InitCloudProvider needs to know whether `cloudProvidersDisabled` is true or false and your method signature needs to reflect that..

Reply:
This is fixed now thanks to @andrewsykim suggestion